spyder/plugins/updatemanager/workers.py

```diff
@@ -66,6 +68,73 @@ def _rate_limits(page):
     logger.debug("\n\t".join(msg_items))
 
 
+class UpdateType:
+    """Enum with the different update types."""
+
+    Major = "major"
+    Minor = "minor"
+    Micro = "micro"
+
+
+class AssetInfo(TypedDict):
+    """Schema for asset information."""
+
+    # Filename with extension of the release asset to download.
+    filename: str
+
+    # Type of update
+    update_type: UpdateType
+
+    # Download URL for the asset.
+    url: str
+
+
+def get_asset_info(release: str) -> AssetInfo:
```

Comment:
```suggestion
def get_asset_info(release: str | packaging.version.Version) -> AssetInfo:
```

I didn't realize in my initial review that this can also receive a `packaging` `Version`.

Reply:
How about this?
```suggestion
def get_asset_info(release: str | Version) -> AssetInfo:
```